File: app/src/main/java/com/treelzebub/umap/util/BusProvider.java
package com.treelzebub.umap.util;

import android.util.Log;

import org.jetbrains.annotations.NotNull;

/**
 * Created by Tre Murillo on 5/16/15
 * Copyright(c) 2015
 */
public class BusProvider {
    private static UMapBus BUS;

    @NotNull
    public static UMapBus getInstance() {
        if (BUS == null) {
            Log.d("BusProvider", "Instantiate new provider");
            BUS = new UMapBus();
        }
        return BUS;
    }

    private BusProvider() {
    }
}

File: app/src/main/kotlin/com/treelzebub/umap/util/BusProvider.kt
package com.treelzebub.umap.util

/**
 * Created by Tre Murillo on 5/28/15
 */
public object BusProvider {
    private var bus: UMapBus? = null
    public val instance: UMapBus
        get() = bus ?: UMapBus()
}
